react-components/src/components/RevealToolbar/RuleBasedOutputsButton.tsx

```diff
@@ -22,35 +22,57 @@ import { useTranslation } from '../i18n/I18n';
 import { useFetchRuleInstances } from '../RuleBasedOutputs/hooks/useFetchRuleInstances';
 import { type AssetStylingGroup } from '../..';
 
+type EmptyRule = {
+  rule: any;
+  isEnabled: boolean;
+};
 type RuleBasedOutputsButtonProps = {
   onRuleSetStylingChanged?: (stylings: AssetStylingGroup[] | undefined) => void;
+  onRuleSetSelectedChanged?: (ruleSet: RuleAndEnabled | EmptyRule | undefined) => void;
 };
 export const RuleBasedOutputsButton = ({
-  onRuleSetStylingChanged
+  onRuleSetStylingChanged,
+  onRuleSetSelectedChanged
 }: RuleBasedOutputsButtonProps): ReactElement => {
   const [currentRuleSetEnabled, setCurrentRuleSetEnabled] = useState<RuleAndEnabled>();
-  const [ruleInstances, setRuleInstances] = useState<RuleAndEnabled[]>();
+  const [emptyRuleSelected, setEmptyRuleSelected] = useState<EmptyRule>();
+  const [ruleInstances, setRuleInstances] =
+    useState<Array<RuleAndEnabled | EmptyRule | undefined>>();
   const { t } = useTranslation();
 
   const ruleInstancesResult = useFetchRuleInstances();
 
+  const emptySelection: EmptyRule = {
+    rule: { properties: { id: '', name: t('RULESET_NO_SELECTION', 'No RuleSet selected') } },
+    isEnabled: false
+  };
+
   useEffect(() => {
     if (ruleInstancesResult.data === undefined) return;
 
     setRuleInstances(ruleInstancesResult.data);
   }, [ruleInstancesResult]);
 
-  const onChange = (data: { target: { id: string | number; checked: boolean } }): void => {
+  const onChange = (data: string | undefined): void => {
     ruleInstances?.forEach((item) => {
+      if (item === undefined) return;
       item.isEnabled = false;
     });
+    emptySelection.isEnabled = false;
+
     const selectedRule = ruleInstances?.find((item) => {
-      return item.rule.properties.id === data.target.id && data.target.checked;
+      return item?.rule?.properties.id === data;
     });
 
     if (selectedRule !== undefined) {
-      selectedRule.isEnabled = data.target.checked;
+      selectedRule.isEnabled = true;
+    } else {
+      emptySelection.isEnabled = true;
+      if (onRuleSetStylingChanged !== undefined) onRuleSetStylingChanged(undefined);
     }
+    if (onRuleSetSelectedChanged !== undefined) onRuleSetSelectedChanged(selectedRule);
+
+    setEmptyRuleSelected(emptySelection);
     setCurrentRuleSetEnabled(selectedRule);
     setRuleInstances(ruleInstances);
```

Comment:
I'm not sure this will do what we want (even though I know this is how it was before). But setting the state to the same object as the previous state (even though it has been modified in the meantime), won't re-trigger hooks that have it as dependency (since dependency checks only look at the reference of the dependency, not the content). So I think we need to clone the list before modifying it in this callback.

Reply:
actually I removed.
https://github.com/cognitedata/reveal/pull/4453/commits/5824d886e02f0d1ee6ecdf3c0738b52e885dd19f